packages/blockchain-wallet-v4-frontend/src/data/modals/slice.ts

```diff
@@ -0,0 +1,47 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { merge } from 'ramda'
+
+import { ModalNameType, ModalsState, UpdateModalOptionsPayload, UpdateModalPayload } from './types'
+
+const initialState: ModalsState = []
+
+const modalsSlice = createSlice({
+  initialState,
+  name: 'modals',
+  reducers: {
+    closeAllModals: (state) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      state = []
```

Comment:
This is a small gotcha. When reseting the state completely like you are here, just return the `initialState` instead of trying to modify the existing state. 
`return initialState` 
more info here https://redux-toolkit.js.org/usage/immer-reducers#resetting-and-replacing-state